backend/rates/rates.go

```diff
@@ -241,28 +215,34 @@ func (updater *RateUpdater) lastUpdateLoop(ctx context.Context) {
 }
 
 func (updater *RateUpdater) updateLast(ctx context.Context) {
-	url := fmt.Sprintf(cryptoCompareURL,
-		strings.Join(coins, ","),
-		strings.Join(fiats, ","),
-	)
-	req, err := http.NewRequest("GET", url, nil)
+	param := url.Values{
+		"ids":           {simplePriceAllIDs},
+		"vs_currencies": {simplePriceAllCurrencies},
+	}
+	endpoint := fmt.Sprintf("%s/simple/price?%s", updater.coingeckoURL, param.Encode())
+	req, err := http.NewRequest("GET", endpoint, nil)
 	if err != nil {
-		updater.log.Errorf("updateLast: http.NewRequest: %v", err)
+		updater.log.WithError(err).Error("could not create request")
+		updater.last = nil
 		return
 	}
-	response, err := updater.httpClient.Do(req.WithContext(ctx))
+
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+	res, err := updater.httpClient.Do(req.WithContext(ctx))
```

Comment:
I think you'll want to do it from within `updater.geckoLimiter.Call`. Here's an example: https://github.com/digitalbitbox/bitbox-wallet-app/blob/5ac525c/backend/rates/history.go#L255.

Otherwise, this bypasses the rate limiter.

Reply:
ah again :innocent: fixed!